itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestStatsReplicationScenarios.java

```diff
@@ -301,12 +338,106 @@ private String dumpLoadVerify(List<String> tableNames, String lastReplicationId,
     return dumpTuple.lastReplicationId;
   }
 
+  /**
+   * Run a bootstrap that will fail.
+   * @param tuple the location of bootstrap dump
+   */
+  private void failBootstrapLoad(WarehouseInstance.Tuple tuple, int failAfterNumTables) throws Throwable {
+    // fail setting ckpt directory property for the second table so that we test the case when
+    // bootstrap load fails after some but not all tables are loaded.
+    BehaviourInjection<CallerArguments, Boolean> callerVerifier
+            = new BehaviourInjection<CallerArguments, Boolean>() {
+      int cntTables = 0;
+      @Nullable
+      @Override
+      public Boolean apply(@Nullable CallerArguments args) {
+        cntTables++;
```

Comment:
This stub will be called multiple times per table as it is invoked by several methods in InjectableBehaviourObjectStore. Need to increment the count only if encounter a new table.

Reply:
Hmm. Thanks for catching this. Done.